Use tasklist-task index on Operate

panagiotisgts opened the pull request:
## Description
Make use of the new `tasklist-task` index in Operate's BE services for Elasticsearch and Opensearch

<!-- Describe the goal and purpose of this PR. -->

## Checklist

<!--- Please delete options that are not relevant. Boxes should be checked by reviewer. -->
- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #23891

sdorokhova commented on operate/schema/src/main/java/io/camunda/operate/schema/IndexTemplateDescriptorsConfigurator.java:
```diff
@@ -129,7 +130,15 @@ public SequenceFlowTemplate getSequenceFlowTemplate(
 
   @Bean
-  public TaskTemplate getUserTaskTemplate(final DatabaseInfo databaseInfo) {
+  public UserTaskTemplate getUserTaskTemplate(
```
❔ Why do we still need it? Because of importer? Should we create a follow-up task to fully remove it?

panagiotisgts replied:
@sdorokhova Removed UserTaskTemplate and importers for UserTasks in the followup commit

sdorokhova reviewed (changes requested):
Overall looks good to me!
Left couple of comments in code

ChrisKujawa commented:
@panagiotisgts @sdorokhova what is the status here?

sdorokhova reviewed (approved):
Looks good to me!
Only question: we deleted couple of tests (both unit and IT). Is it because it's covered by some other tests?

panagiotisgts commented:
@sdorokhova 
> Only question: we deleted couple of tests (both unit and IT). Is it because it's covered by some other tests?

The tests that were deleted are related to the User Task importers which were also removed. They are covered by Tasklists 8.6 importer and the CamundaExporter tests as well.
